src/CraftingRecipes.cpp

```diff
@@ -874,3 +877,182 @@ void cCraftingRecipes::HandleFireworks(const cItem * a_CraftingGrid, cCraftingRe
 
 
 
+
+void cCraftingRecipes::HandleDyedLeather(const cItem * a_CraftingGrid, cCraftingRecipes::cRecipe * a_Recipe, int a_GridStride, int a_GridWidth, int a_GridHeight)
+{
+	short result_type = a_Recipe->m_Result.m_ItemType;
+	if ((result_type == E_ITEM_LEATHER_CAP) || (result_type == E_ITEM_LEATHER_TUNIC) || (result_type == E_ITEM_LEATHER_PANTS) || (result_type == E_ITEM_LEATHER_BOOTS))
+	{
+		bool found = false;
+		cEnchantments enchants;
+		AString name;
+		AString lore;
+		short damage = 0;
+		int repair = 0;
+
+		float red = 0;
+		float green = 0;
+		float blue = 0;
+		float dye_count = 0;
+
+		for (int x = 0; x < a_GridWidth; ++x)
+		{
+			for (int y = 0; y < a_GridHeight; ++y)
+			{
+				int GridIdx = x + a_GridStride * y;
+				if ((a_CraftingGrid[GridIdx].m_ItemType == result_type) && (found == false))
+				{
+					found = true;
+					enchants = a_CraftingGrid[GridIdx].m_Enchantments;
+					name = a_CraftingGrid[GridIdx].m_CustomName;
+					lore = a_CraftingGrid[GridIdx].m_Lore;
+					damage = a_CraftingGrid[GridIdx].m_ItemDamage;
+					repair = a_CraftingGrid[GridIdx].m_RepairCost;
```

Comment:
This is not a good way. If there ever is an update that adds yet another thing to `cItem`, this will need updating. Why don't you instead make a copy of the entire `cItem` object.